client/client_pool/src/external_client.cpp

```diff
@@ -264,19 +271,25 @@ BaseCommConfig* ConcordClient::CreateCommConfig() const {
                           pool_config_.use_unified_certificates,
                           nullptr,
                           secretData};
+#endif
+  throw std::runtime_error("FOR TLS COMM SHOULD HAVE BEEN COMPILED WITH \"USE_COMM_TLS_TCP\"");
 }
 
 SharedCommPtr ConcordClient::ToCommunication(const BaseCommConfig& comm_config) {
   if (comm_config.commType_ == TlsTcp) {
+#ifdef USE_COM_TLS_TCP
```

Comment:
A better approach:
use a BaseCommConfig*, initialize it in the body of the runtime ifs and wrap the ifs with an ifdef.

Reply:
There's a follow up task to make order in lib communication

Reply:
this is a small modification which relates to the other changes you make in this branch, so I see no reason not to make this specific modification in this pull request